### PR TITLE
New Rule: id-name-format

### LINKED ISSUE #325: New Rule: ID Name Format
Add rule to verify name format of id selectors.

Please see the original issue for more information and discussion #17 and #275

### PR DESCRIPTION
**Pull request by benthemonkey**
Clone of `class-name-format`, although I can't imagine why a person would want to use suffix extensions on an ID.

Will close #325 

DCO 1.1 Signed-off-by: Ben Rothman <bensrothman@gmail.com>

### CODE REVIEW

File: docs/rules/id-name-format.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+#snake_case {
+  content: '';
+}
+
+#foo {
+  @extend #camelCase;
+}
+```
```
Comment (DanPurdy) [inline docs/rules/id-name-format.md]: A quick example of using the ignore property would be good here.

**Comment by DanPurdy** [comment]
Just a wuick one for the docs other than that :+1:

**Comment by benthemonkey** [comment]
@DanPurdy done.

**Comment by DanPurdy** [comment]
:+1:

**Pull request merged**